### DISCUSSION TITLE
What versioning standard spacy uses?

### DISCUSSION THREAD

**Discussion by dhgoratela** [explosion/spaCy#9865] [Help: Other Questions] [answered]
I would like to know what versioning standard spacy uses for its releases. Is it [Semantic Versioning](https://semver.org)?
If yes, the release [v3.2.0](https://github.com/explosion/spaCy/releases/tag/v3.2.0)  has **backward incompatibility** changes, which calls for being labeled as a **Major release**. Instead, the **Minor** version number has been incremented (3.1.4 => 3.2.0). 
Can anyone please help me understand?

**Comment by adrianeboyd** [comment, marked as answer]
Are you running into problems using spacy due to the upgrade to v3.2.0 or are you just asking a question out of curiosity?

Basically, the header in the release notes is not quite accurate, but we want to present this information in a way that users will notice. In this case, it mainly means "changes in behavior" rather than "backwards API incompatibilities". The changes to the language-specific tokenizer APIs are actually backwards incompatible, but the tokenizer init is mostly internals (no changes to any docs for https://spacy.io are needed) and this brings these APIs in line with other similar classes in spacy.

**Reply by dhgoratela** [reply]
Thanks for the answer. Question was for generally understanding the method spaCy follows to label its releases. The Semantic versioning says that backward incompatibilities need to be marked by major release. Now that you clarified, that those were changes in internal behavior and not with the User API, then it is legit to mark it as Minor release. I am trying to learn.. Thank you for the explaination..
